### DISCUSSION TITLE
Using sqlite as database

### DISCUSSION THREAD

**Discussion by huangjiahua** [ghostfolio/ghostfolio#1735] [Ideas]
In self hosting environments, using sqlite is often easier and more resource efficient. I know currently Prisma doesn't support JSON type for sqlite, but I'm curious if it's possible to put sqlite support on your roadmap. Thank you!

**Comment by dtslvr** [comment]
Related issue at prisma: https://github.com/prisma/prisma/issues/3786

**Reply by dtslvr** [reply]
`json` and `enum` fields are now supported for _SQLite_ in the latest Prisma release (see https://github.com/prisma/prisma/releases/tag/6.2.0). Feel free to give it a try and let us know how it goes!

**Reply by 403-html** [reply]
TL:DR – there needs to be done some coding/adjusting app, maybe someone will pick this up or maybe I'll give it a try as seems that for sqlite simple converting to a String field and store permissions as a comma-separated string (pseudo-array and managing it in-app) would be enough. But different connectors should be recognized, it should be adjusted for it etc.

hi, I just noticed this thread – you mean ghostfolio could be OOTB compatible with sqlite, and just pointing out to it would be enough as project uses `"@prisma/client": "6.7.0",`?

Because when we change env to point to file (for example `DATABASE_URL="file:./db/portfolio.db"`), then adjust connector to sqlite in `prisma/prisma.scheme` then when we do `npx prisma generate` we will get `error: Field "permissions" in model "Access" can't be a list. The current connector does not support lists of primitive types.`
